src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/BlazorDebugConfigurationProvider.ts

```diff
@@ -103,12 +103,13 @@ export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurati
     }
 
     private async launchBrowser(folder: vscode.WorkspaceFolder | undefined, configuration: vscode.DebugConfiguration) {
+        const setNoDebugFlag = (url: string) => configuration.noDebug ? `${url}?_blazor_debug=false` : url;
         const browser = {
             name: JS_DEBUG_NAME,
             type: configuration.browser === 'edge' ? 'pwa-msedge' : 'pwa-chrome',
             request: 'launch',
             timeout: configuration.timeout || 30000,
-            url: configuration.url || 'https://localhost:5001',
+            url: setNoDebugFlag(configuration.url || 'https://localhost:5001'),

```

Comment:
If we don't enforce `configuration.url` not to have query-strings already this will break, right? If we enforce that already great, if not you might have to use a URL class/function to add the querystring in a more robust way.

Reply:
This change should be backwards compatible. If the query param is set but the framework doesn't contain the matching updates, then the flag will no-op.

If it is not set and the framework is updated then it defaults to true.

Reply:
Oh, I meant that if `configuration.url = 'https://localhost:5001?somequeryString=stuff'` then this new function does `url:`${configuration.url}?_blazor_debug=false`` then we end up with `https://localhost:5001?somequeryString=stuff?_blazor_debug=false`, which would result in some weird querystrings.

Reply:
Ah I see. We decided to keep the logic simple here. We don't anticipate that a lot of people will be launching with custom query strings. If they are, it's fairly easy to work around (e.g., edit the URL)/